src/utils/getIssueAntdPath.ts

```diff
@@ -17,21 +17,37 @@ const serializePath = (
   const baseSchema = getSchemaBaseSchema(schema);
   if (isZodObject(baseSchema)) {
     antdPath.push(subPath);
-    const subPathBaseSchema = getSchemaBaseSchema(baseSchema.shape[subPath]);
 
-    return isZodObject(subPathBaseSchema)
-      ? serializePath(subPathBaseSchema, zodPath, antdPath)
-      : concatAntdPath(antdPath);
+    const replacedPath = subPath.replace(/\.\d+/, "");
+    const subPathBaseSchema = getSchemaBaseSchema(
+      baseSchema.shape[replacedPath]
+    );
+
+    if (isZodObject(subPathBaseSchema))
+      return serializePath(subPathBaseSchema, zodPath, antdPath);
+    if (isZodArray(subPathBaseSchema))
+      return serializePath(subPathBaseSchema.element, zodPath, antdPath);
+
+    return concatAntdPath(antdPath);
   }
 
   throw new Error("Unknown zod schema");
 };
 
 const getIssueAntdPath = (schema: ZodTypeAny, issue: ZodIssue) => {
-  const zodPath = issue.path.reverse() as string[];
+  // issue.path = ["testArray", "0", "myField"] => modifiedPath = ["testArray.0", "myField"]
+  const modifiedPath = issue.path.reduce<string[]>((prev: string[], curr) => {
+    if (typeof curr === "number") {
```

Comment:
Why is here a specific condition checking for `number`?

Reply:
because when we are dealing with antd form lists paths have this structure: arrayName.index.anythingElse but zod cant understand the index part so it is necessary to join arrayName.index as a single section of the path hence I need to check for a number that would be the index

Reply:
Thanks for clarification. Let's do it like this.

Extract this reduce to a new helper function called `prepareAntPath(path: string[])`.
 It can be a part of the same file.

Eventually it should look like:
```
const getIssueAntdPath = (schema: ZodTypeAny, issue: ZodIssue) => {
  const preparedPath = prepareAntPath(issue.path);
  ...
}
```